File: 2026/07-July/28-Wednesday/solution.cs
using System;

public class Solution {
    public string SmallestPalindrome(string s) {
        int n = s.Length;
        int halfLen = n / 2;

        // Extract and sort the first half
        char[] half = s.Substring(0, halfLen).ToCharArray();
        Array.Sort(half);

        string leftHalf = new string(half);
        string middle = (n % 2 != 0) ? s[halfLen].ToString() : "";

        // Reverse the sorted half to get the right side
        Array.Reverse(half);
        string rightHalf = new string(half);

        return leftHalf + middle + rightHalf;
    }
}

File: 2026/07-July/28-Wednesday/solution.java
import java.util.Arrays;

class Solution {
    public String smallestPalindrome(String s) {
        int n = s.length();
        int halfLen = n / 2;

        // Extract the first half and sort its characters
        char[] half = s.substring(0, halfLen).toCharArray();
        Arrays.sort(half);

        String leftHalf = new String(half);
        String middle = (n % 2 != 0) ? String.valueOf(s.charAt(halfLen)) : "";

        // Reverse the sorted half for the right side
        StringBuilder sb = new StringBuilder(leftHalf);
        String rightHalf = sb.reverse().toString();

        return leftHalf + middle + rightHalf;
    }
}
